api/client.go

```diff
@@ -68,14 +71,23 @@ func NewClient(cfg Config) (Client, error) {
 	}
 	u.Path = strings.TrimRight(u.Path, "/")
 
+	var h map[string]string
+	if cfg.Headers != nil {
+		h = cfg.Headers
+	} else {
+		h = map[string]string{}
+	}
+
 	return &httpClient{
 		endpoint: u,
+		headers:  h,
```

Comment:
I guess you can just write `headers: cfg.Headers,` here and don't need the indirection via `h`.